### PR TITLE
Comparison operations implementation

### PR DESCRIPTION
**Pull request by keithschulze**
Addresses #82. Initial implementations for `clamp`, `cmp`, `eif`, `lt`, `le`, `gt`, `ge`, `ne` and `eq`.

### CODE REVIEW

File: src/main/clojure/clojure/core/matrix.clj [hunk 1]
```diff
@@ -1701,6 +1701,119 @@
   ([m]
     (mp/element-max m)))
 
+(defn clamp
+  "Clamps each element in a numerical array between an upper and lower bound.
+  Returns an error if the lower bound is greater that the upper bound.
```
Comment (mikera) [inline src/main/clojure/clojure/core/matrix.clj:1706]: I would not guarantee an error here in the docstring. could be implementation dependent.

**Comment by mikera** [comment]
Looks really good, thanks for working on this! I think there are a few things to tidy up still but will merge and have a play with it.

File: src/main/clojure/clojure/core/matrix.clj (hunk 1 above)
Comment (keithschulze) [inline src/main/clojure/clojure/core/matrix.clj:1706]: Yes, you're right. I'll remove it.

Comment (keithschulze) [inline src/main/clojure/clojure/core/matrix.clj:1706]: I also put that in the spec of the element-clamp protocol:

```
(element-clamp [m a b] 
    "Returns a matrix where the elements are clamped to be within upper and 
    lower bounds specified by a and b. Must throw an exception if a > b.")
```

Should I remove it from there too? I will change "upper and lower bounds" to "lower and upper bounds" as it's not consistent with the definition.

Comment (mikera) [inline src/main/clojure/clojure/core/matrix.clj:1706]: Yeah I think better to leave unspecified. Testing for special cases and throwing exceptions can expensive, some implementations will not want to do it.

**Comment by mikera** [comment]
Just pushed a few extra changes.

**Comment by keithschulze** [comment]
Changes look great. I like addition of extra arities.

**Pull request merged**